File: 03_Basic_Recursion/04-A_Print_Sum_First_NNumbers.cs
// parameterised way

Console.Write("Enter value of N: ");
int N = Convert.ToInt32(Console.ReadLine());

f(N, 0);

void f(int i, int sum)
{
    if(i < 1)
    {
        Console.WriteLine(sum);
        return;
    }
    
    f(i-1, sum + i);
}

File: 03_Basic_Recursion/04-A_Print_Sum_First_NNumbers.cpp
#include <bits/stdc++.h>
using namespace std;

void f(int i, int sum)
{
    if (i < 1)
    {
        cout << sum << endl;
        return;
    }

    f(i - 1, sum + i);
}

int main()
{
    int N;
    cout << "Enter value of N: ";
    cin >> N;

    f(N, 0);

    return 0;
}